quartz_solar_forecast/forecasts/v2.py

```diff
@@ -0,0 +1,234 @@
+import datetime
+import pandas as pd
+import zipfile
+import gdown
+import os.path
+
+from quartz_solar_forecast.weather import WeatherService
+
+from xgboost.sklearn import XGBRegressor
+
+
+class TryolabsSolarPowerPredictor:
+    """
+    A class to predict solar power output based on weather data, location, panel orientation,
+        and other factors.
+
+    Attributes
+    ----------
+    model_path : str
+        Path to the trained model joblib file.
+
+    Methods
+    -------
+    load_model -> None:
+
+        Downloads the model from Google Drive and decropresses it if necessary
+    
+    predict_power_output(latitude: float, longitude: float, start_date: str, kwp: float,
+        orientation: float, tilt: float) -> pd.DataFrame:
+
+        Predicts solar power output for the given parameters.
+    """
+    DATE_COLUMN = "date"
+
+    def _download_model(self, filename: str, file_id: str) -> None:
+        """
+        Download model from google drive.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the model to be saved
+        file_id: 
+            Google id of the model file
+        """
+        gdown.download(f'https://drive.google.com/uc?id={file_id}', filename, quiet=False)
```

Comment:
where does the model get saved?

Reply:
I added it to the description in the doc string. I would keep the location and keep the forecast folder for scripts.